commit 1fcaeb014b639d80e14d74abc27fae63f159d9d1
Author: Jerome Brunet <jbrunet@baylibre.com>
Date:   Fri Jan 20 08:20:24 2017 -0800

    ARM64: dts: meson-gxbb-odroidc2: fix GbE tx link breakage
    
    
    [ Upstream commit feb3cbea0946c67060e2d5bcb7499b0a6f6700fe ]
    
    OdroidC2 GbE link breaks under heavy tx transfer. This happens even if the
    MAC does not enable Energy Efficient Ethernet (No Low Power state Idle on
    the Tx path). The problem seems to come from the phy Rx path, entering the
    LPI state.
    
    Disabling EEE advertisement on the phy prevent this feature to be
    negociated with the link partner and solve the issue.
    
    Signed-off-by: Jerome Brunet <jbrunet@baylibre.com>
    Signed-off-by: Kevin Hilman <khilman@baylibre.com>
    Signed-off-by: Arnd Bergmann <arnd@arndb.de>
    Signed-off-by: Sasha Levin <alexander.levin@verizon.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/arch/arm64/boot/dts/amlogic/meson-gxbb-odroidc2.dts b/arch/arm64/boot/dts/amlogic/meson-gxbb-odroidc2.dts
index e6e3491..f150a4c 100644
--- a/arch/arm64/boot/dts/amlogic/meson-gxbb-odroidc2.dts
+++ b/arch/arm64/boot/dts/amlogic/meson-gxbb-odroidc2.dts
@@ -85,6 +85,18 @@
 	status = "okay";
 	pinctrl-0 = <&eth_pins>;
 	pinctrl-names = "default";
+	phy-handle = <&eth_phy0>;
+
+	mdio {
+		compatible = "snps,dwmac-mdio";
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		eth_phy0: ethernet-phy@0 {
+			reg = <0>;
+			eee-broken-1000t;
+		};
+	};
 };
 
 &ir {
